core/src/main/scala/io/km8/core/kafka/KafkaExplorer.scala

```diff
@@ -45,20 +46,17 @@ object KafkaExplorer {
     def withAdminClient[A](clusterId: String)(eff: AdminClient => RIO[Blocking with Clock, A]): Task[A] =
       ZIO
         .service[AdminClient]
-        .flatMap { ac =>
-          eff(ac)
-            .timeoutFail(new Exception("Timed out"))(6.seconds)
-        }
+        .flatMap(eff(_).timeoutFail(new Exception("Timed out"))(6.seconds))
         .provideLayer(blockingLayer >+> adminClientLayer(clusterId) ++ clockLayer)
 
     override def listBrokers(clusterId: String) =
       withAdminClient(clusterId) { ac =>
         for {
-          (nodes, controllerId) <- ac.describeClusterNodes() <&> ac.describeClusterController().map(_.id)
+          (nodes, controllerId) <- ac.describeClusterNodes() <&> ac.describeClusterController().map(_.map(_.id))
           brokers = nodes.map { n =>
                       val nodeId = n.id
-                      if (controllerId != nodeId) BrokerDetails(id = nodeId, isController = false)
-                      else BrokerDetails(nodeId, isController = true)
+                      if (controllerId.contains(nodeId)) BrokerDetails(nodeId, isController = true)
+                      else BrokerDetails(id = nodeId, isController = false)
                     }
           // resources = nodes.map(n => new ConfigResource(ConfigResource.Type.BROKER, n.idString()))
```

Comment:
do we need this comment?

Reply:
I kept it there because that call might bring more details about brokers, will keep it there a bit more as I have an idea where to use it.